### PR TITLE
Prepare data for work with colors

### PR DESCRIPTION
**Pull request by decobots**

### CODE REVIEW

File: src/color_table.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+class Color:
+    def __init__(self, r: int, g: int, b: int):
+        self.__rgb_checker(r, g, b)
+        self.rgb = colormath.color_objects.AdobeRGBColor(rgb_r=r, rgb_g=g, rgb_b=b)
+        self.lab = colormath.color_conversions.convert_color(
+            color=self.rgb,
+            target_cs=colormath.color_objects.LabColor)
+
+    @staticmethod
+    def __rgb_checker(r, g, b):
```
Comment (livercat) [inline src/color_table.py]: Rename to a verb (for example, "__check_rgb")

File: src/color_table.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+                             ]
+        return sorted(calculated_deltas, key=lambda x: list(x.values())[0])
+
+    def __generate_files_with_color_tables(self):
+        rgb_colors = {}
+        lab_colors = {}
+        with open(file=self.file_raw, mode='r') as inp, \
+                open(file=self.file_lab, mode="w") as out_lab, \
+                open(file=self.file_rgb, mode="w") as out_rgb:
+            for line in inp:
```
Comment (livercat) [inline src/color_table.py]: You can use "for line in inp.splitlines()" to avoid stripping "\n" later

File: src/color_table.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+                line = line.rstrip('\n')
+                name, r, g, b = line.split('	')
+                c = Color(int(r), int(g), int(b))
```
Comment (livercat) [inline src/color_table.py]: c -> color

Comment (decobots) [reply src/color_table.py]: done

File: src/color_table.py [hunk 4]
```diff
@@ -41,9 +42,9 @@ def __init__(self,
                  filename_raw_data: str = 'colors.txt',
                  filename_lab_out: str = 'out_lab.txt',
                  filename_rgb_out: str = 'out_rgb.txt'):
-        self.file_raw = filename_raw_data
-        self.file_lab = filename_lab_out
-        self.file_rgb = filename_rgb_out
+        self.file_raw = os.path.join(os.path.dirname(os.path.dirname(__file__)), filename_raw_data)
```
Comment (livercat) [inline src/color_table.py]: Don't do path operations inside this class - move them to the call site

File: tests/test_color_table.py [hunk 5]
```diff
@@ -21,7 +22,25 @@ def test_lab_values():
 
 
 def test_color_table():
-    test_table = ColorTable(filename_raw_data="colors_test.txt",
-                            filename_lab_out="lab_test_txt",
-                            filename_rgb_out="rgb_test.txt")
-
+    with suppress(FileNotFoundError):
```
Comment (livercat) [inline tests/test_color_table.py]: This should be in setup

File: tests/test_color_table.py [hunk 6]
```diff
@@ -21,7 +22,25 @@ def test_lab_values():
...
+        os.remove(os.path.join(os.path.dirname(os.path.dirname(__file__)), "tests/lab_test_txt"))
+    with suppress(FileNotFoundError):
+        os.remove(os.path.join(os.path.dirname(os.path.dirname(__file__)), "tests/rgb_test_txt"))
+    test_table = ColorTable(filename_raw_data=os.path.join("tests", "colors_test.txt"),
```
Comment (livercat) [inline tests/test_color_table.py]: Looks like a fixture

File: src/color_table.py (hunk 4 above)
Comment (decobots) [reply src/color_table.py]: ok

File: tests/test_color_table.py (hunk 5 above)
Comment (decobots) [reply tests/test_color_table.py]: renamed test

File: tests/test_color_table.py (hunk 6 above)
Comment (decobots) [reply tests/test_color_table.py]: It used only in that test

File: src/color_table.py (hunk 1 above)
Comment (decobots) [reply src/color_table.py]: done

**Pull request merged**